[#61306] Dialog text should use "Cancel occurrence"

myabc opened the pull request:
# Ticket

https://community.openproject.org/wp/61306

# What are you trying to accomplish?

Updates the title, description and confirmation button text of `Meetings::DeleteDialogComponent` to use the term "cancel" rather than "delete" for removing occurrences.

**Danger Dialogs for cancelling open/instantiated occurrences and planned/scheduled occurrences should now be visually identical.**

## Screenshots

| Scenario                      | Before | After |
|-------------------------------|--------|--------|
| open/instantiated occurrences | <img width="519" alt="Screenshot 2025-02-12 at 11 54 21" src="https://github.com/user-attachments/assets/20ef4cf1-93e1-433b-9f21-bad441e83a98" /> | <img width="507" alt="Screenshot 2025-02-12 at 11 54 54" src="https://github.com/user-attachments/assets/70ff1ce5-1e1c-476d-b3b9-faa0e69aee57" /> |
| planned/scheduled occurrences | <img width="502" alt="Screenshot 2025-02-12 at 11 54 30" src="https://github.com/user-attachments/assets/2cc7333b-9818-40cc-9010-ba3ea3a62521" /> | <img width="495" alt="Screenshot 2025-02-12 at 11 55 04" src="https://github.com/user-attachments/assets/b3c071aa-c04b-4bda-8cc1-f0f6b4d98d48" /> | 

# What approach did you choose and why?

I stuck with the 1-to-1 controller action to dialog approach introduced in #17569. It is not the DRY-est approach, but it reduces complexity. It might be possible to make `Meetings::DeleteDialogComponent` polymorphic in the future - thereby allowing us to drop `RecurringMeetings::ScheduledDeleteDialogComponent`.

Additionally, I've kept the I18n string keys separate for now. Although we are making the UI for cancelling open/instantiated occurrences and planned/scheduled occurrences visually identical for moment, I could envisage us making adjustments to the copy in the future.

# Merge checklist

- [X] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc commented:
> Additionally, I've kept the I18n string keys separate for now. Although we are making the UI for cancelling open/instantiated occurrences and planned/scheduled occurrences visually identical for moment, I could envisage us making adjustments to the copy in the future.

I might be wrong here - let me know what you think.

mrmir commented:
I just took a quick look here, but FYI we don't need to update strings for all locales I don't think. Only the English ones are necessary as the rest get overwritten by the crowdin action anyway

myabc commented:
> I just took a quick look here, but FYI we don't need to update strings for all locales I don't think. Only the English ones are necessary as the rest get overwritten by the crowdin action anyway

I wasn't sure if Crowdin synced both ways. I'll update the PR to only touch `en.yml`.

myabc committed:
[#61306] Dialog text should use "Cancel occurrence"

Danger Dialog title, description and confirmation button text should
refer to "cancelling" not "deleting" the occurrence - Danger Dialogs for
cancelling open/instantiated occurrences and planned/scheduled
occurrences should now be visually identical.
